### PR TITLE
i18n(ko-KR): update `middleware.mdx`, `routing.mdx`, `api-reference.mdx`

### PR DESCRIPTION
**Pull request by jsparkdev**
#### Description (required)

- update `middleware.mdx`
- update `routing.mdx`
- update `api-reference.mdx`
- #8967
- #8994
- #8993
- #8943
- #9006

#### Related issues & labels (optional)

- Closes #
- Suggested label:

### CODE REVIEW

**Review by dreyfus92 — APPROVED** [review]
LGTM ✅

File: src/content/docs/ko/guides/middleware.mdx [hunk 1]
```diff
@@ -196,6 +197,79 @@ auth 응답
...
+<p><Since v="4.13.0" /></p>
+
+`APIContext`는 `rewrite()`라는 메서드를 노출하는데, 이는 [Astro.rewrite](/ko/guides/routing/#리라이트)와 동일한 방식으로 작동합니다.
+
+방문자를 새 페이지로 [리디렉션](/ko/guides/routing/#동적-리디렉션)하지 않고 다른 페이지의 콘텐츠를 표시하려면 미들웨어에서 `context.rewrite()`를 사용하세요. 이렇게 하면 새 렌더링 단계가 트리거되어 모든 미들웨어가 다시 실행됩니다.
+
+```js title="src/middleware.js"
+import { isLoggedIn } from "~/auth.js"
+export function onRequest (context, next) {
+  if (!isLoggedIn(context) {
```
Comment (liruifengv) [inline src/content/docs/ko/guides/middleware.mdx:211]:
```suggestion
  if (!isLoggedIn(context)) {
```

I fixed this in https://github.com/withastro/docs/pull/8994

File: src/content/docs/ko/guides/middleware.mdx [hunk 2]
```diff
@@ -196,6 +197,79 @@ auth 응답
...
+  return next();
+};
+```
+
+또한 `next()` 함수에 선택적 URL 경로 매개변수를 전달하여 새 렌더링 단계를 다시 트리거하지 않고 현재 `Request`를 리라이트할 수 있습니다. 리라이트 경로의 위치는 문자열, URL 또는 `Request`로 제공할 수 있습니다:
+
+```js title="src/middleware.js"
+import { isLoggedIn } from "~/auth.js"
+export function onRequest (context, next) {
+  if (!isLoggedIn(context) {
```
Comment (liruifengv) [inline src/content/docs/ko/guides/middleware.mdx:231]:
```suggestion
  if (!isLoggedIn(context)) {
```

File: src/content/docs/ko/guides/routing.mdx [hunk 3]
```diff
@@ -298,6 +298,68 @@ if (!isLoggedIn(cookie)) {
...
+
+<p><Since v="4.13.0" /></p>
+
+리라이트를 사용하면 브라우저를 다른 페이지로 리디렉션하지 않고도 다른 경로를 제공할 수 있습니다. 브라우저는 URL 표시줄에 원래 주소를 표시하지만, 대신 [`Astro.rewrite()`](/ko/reference/api-reference/#astrorewrite)에 제공된 URL의 콘텐츠를 표시합니다.
+
+:::tip
+영구적으로 이동한 콘텐츠를 제공하거나 사용자를 새 URL을 사용하는 다른 페이지로 안내하려면 (예: 로그인 후 사용자 대시보드) [리디렉션](#리디렉션)을 사용하세요.
+:::
+
+리라이트는 두 개의 다른 소스 파일을 유지 관리할 필요 없이 여러 경로 (예: `/products/shoes/men/` 및 `/products/men/shoes`/)에서 동일한 콘텐츠를 표시하는 데 유용할 수 있습니다.
```
Comment (liruifengv) [inline src/content/docs/ko/guides/routing.mdx:311]:
```suggestion
리라이트는 두 개의 다른 소스 파일을 유지 관리할 필요 없이 여러 경로 (예: `/products/shoes/men/` 및 `/products/men/shoes/`)에서 동일한 콘텐츠를 표시하는 데 유용할 수 있습니다.
```

Fixed the path typo in https://github.com/withastro/docs/pull/8993

Comment (jsparkdev) [reply src/content/docs/ko/guides/routing.mdx:311]: Thank you!

File: src/content/docs/ko/guides/middleware.mdx (hunk 1 above)
Comment (jsparkdev) [reply src/content/docs/ko/guides/middleware.mdx:211]: Thanks!

File: src/content/docs/ko/guides/middleware.mdx (hunk 2 above)
Comment (jsparkdev) [reply src/content/docs/ko/guides/middleware.mdx:231]: Thank you

**Comment by yanthomasdev** [comment]
Thanks! We're missing this new commit here, could you add it to the PR? https://github.com/withastro/docs/commit/82b728323fdacb08eb35628bd8e41963773a8880

**Comment by jsparkdev** [comment]
@yanthomasdev Sure, I am working on it.

**Review by yanthomasdev — APPROVED** [review]
LGTM! Amazing work! 🙌

**Pull request merged**